### DISCUSSION TITLE
Automatically create a release pull request by triggering a workflow

### DISCUSSION THREAD

**Discussion by toby-thanathip** [nimblehq/android-templates#429] [RFCs]
As of now, when it's time to create a release, we execute the following steps:

1. Create a release PR (manually) from `develop`, then merge it to `master`
2. Draft a new release (automatically, after merging to `master`) → `draft-new-release.yml`
   -> This was integrated in #100 
3. Create a bump version PR (automatically, after triggering the workflow manually) → `bump_version.yml`
   -> This was integrated in #301 

As steps 2 & 3 are automated, so should we do the same for step 1 🚀 

My idea is as follows:

1. Create a new workflow `create_release.yml`
2. Provide user input for the _milestone_ (similar to what we did for `bump_version.yml`)
3. Use https://github.com/octokit/request-action to call GitHub API inside the workflow
4. Call [List repository issues](https://docs.github.com/en/rest/issues/issues?apiVersion=2022-11-28#list-repository-issues), with _milestone_  parameter
5. Parse, filter, map, and organize the response with Kscript (similar to what we did for `draft-new-release.yml`)
6. Create a pull request with the output from our script (similar to what we did for `bump_version.yml`)

**Comment by lydiasama** [comment]
@Tuubz I have a question about how we trigger this workflow. My understanding is that we will trigger it manually from the github repository when the `develop` branch is ready to merge to `main`, not to set it to run this workflow automatically. So I voted `Yes`. Please correct me if I missed something. 🙏🏻

**Reply by toby-thanathip** [reply]
Yes, that's correct! Thank you, for pointing that out 🙏 

It will be triggered manually from GitHub, with _milestone_ as the input, to automatically create the PR  🚀

**Reply by chornerman** [reply]
Voted "Yes" 🚀

**Comment by luongvo** [comment]
I want to continue on this idea to have a new workflow to `automate the release PR creation` step. Do you want to continue on this or delegate to someone else @toby-thanathip?

**Reply by toby-thanathip** [reply]
Please take it 🙏 😂

**Comment by luongvo** [comment]
Ticket is created https://github.com/nimblehq/android-templates/issues/525
